extensions/amp-auto-ads/0.1/placement.js

```diff
@@ -14,19 +14,32 @@
  * limitations under the License.
  */
 
+import {LayoutMarginsChangeDef} from '../../../src/layout-rect';
 import {Services} from '../../../src/services';
+import {clamp} from '../../../src/utils/math';
 import {
   closestByTag,
   createElementWithAttributes,
   scopedQuerySelectorAll,
 } from '../../../src/dom';
+import {computedStyle} from '../../../src/style';
 import {dev, user} from '../../../src/log';
 import {dict} from '../../../src/utils/object';
 import {getAttributesFromConfigObj} from './attributes';
+import {isExperimentOn} from '../../../src/experiments';
 
 /** @const */
 const TAG = 'amp-auto-ads';
 
+/**
+ * @typedef {{
+ *   width: {number|undefined},
+ *   height: {number|undefined},
+ *   margins: {LayoutMarginsChangeDef},
+ * }}
+ */
+let PlacementDef;
```

Comment:
I think something like PlacementSizingDef might be a better name. PlacementDef makes it sound like this might contain all the info that defines a placement (which it doesn't).

Reply:
Done